### DISCUSSION TITLE
VS-Code Extensions - Global Activation ?

### DISCUSSION THREAD

**Discussion by sado17** [microsoft/vscode-discussions#187] [Extension Development QnA] [answered]
Hello,

how I can choose between extensions which are globally activated or other methods in VS Code ?
Are they from beginning globally activated while they will been installed ?

When you ask yourself what I mean, look for example at the extensions LanguagePacks.

Greets
sado

**Comment by alefragnani** [comment, marked as answer]
Hi @sado17 ,

Extensions are activated depending on the **Activation Events** (https://code.visualstudio.com/api/references/activation-events). There are a few _categories_ of activation, and depending on the kind/purpose of your extension, you may choose the one that better fit your needs. 

Globally activated, I suppose _right after VS Code starts, no matter what the user does_. In this case, you should choose `onStartupFinished`. If you want to activate only when the user runs some command, use `onCommand`. And so on..

LanguagePack are a special kind of extension, and doesn't use Activation Events. Anyway,  don't see anyone other than MS publishing it, even existing such option in the Yeoman generator.

Hope this helps

**Reply by sado17** [reply]
Hello, thank you for your answer.
But the globally activated option has only the purpose like the environment variables, system and user, system-wide or user-wide, in this case of vscode, app-internally-global or app-extension-globally, can I understand it in this way ?

Greets
sado

**Reply by alefragnani** [reply]
Extensions are installed as `user-wide`, in `%USERDATA%/.vscode/extensions`. So, if you have two accounts on the same computer, every user will have its own extensions. 

Also, every extension you install is available for every VS Code session you open. If you disable/uninstall some extension in one of the sessions, it will be disabled/uninstalled from the other sessions, when you reload them. You can't have different set of extensions per window, by default, for now... se below.

However, the VS Code team is delivering a new feature called [Settings Profile](https://code.visualstudio.com/updates/v1_72#_settings-profiles) (still in preview stage), which will allow a fine grained control of your extensions, beyond other cool things, and you will be able to have different set of extensions for each Profile. If this is what you want (as a user), I suggest you give it a try.

Hope this helps
